src/index.js

```diff
@@ -1,36 +1,103 @@
+let todos = [];
+const toLocalStorage = () => {
+  localStorage.setItem('todos', JSON.stringify(todos));
+};
+
+const renderTodos = () => {
+  todosListEl.innerHTML = '';
+  if (todos) {
+    todos.forEach((todo, index) => {
+      todosListEl.innerHTML += `
+      <div class="todo" id="${index}">
+        <div class="left-side">
+          <input type="checkbox" id="checkbox" class="checkbox">
+          <p class="task-input-field">${todo.description}</p>
+        </div>
+        <div class="right-side">
+          <i class="bi bi-three-dots-vertical edit"></i>
+          <i class="bi bi-trash delete"></i>
+        </div>
+      </div>
+      `;
+    });
 
-const listArray = [
-  {
-    description: 'wash the dishes',
-    completed: true,
-    index: 1,
-  }, {
-    description: 'complete To Do list project',
-    completed: true,
-    index: 2,
-  },
-  {
-    description: 'wash my car',
-    completed: true,
-    index: 3,
-  },
-];
-const showTasks = (task) => {
-  listArray.sort((a, b) => a.index - b.index);
-  todoList.innerHTML += `<div class="todo" id="todo-con">
-                          <div class="left-side">
-                            <input type="checkbox" id="todo" name="todo" value="">
-                            <p>${listArray[task].description}</p>
-                          </div>
-                          <div class="right-side">
-                            <i class="fa-solid fa-ellipsis-vertical"></i>
-                            <i class="fa-regular fa-trash-can trash-can"></i>
-                          </div>
-                        </div>`;
+    // Function to handle edit action
+    const handleEdit = (index) => {
+      // Get the task description element
+      const taskDescEl = document.getElementsByClassName('task-input-field')[index];
+      // Replace the task description element with an input field
+      const newInputEl = document.createElement('input');
+      newInputEl.setAttribute('type', 'text');
+      newInputEl.setAttribute('value', taskDescEl.innerText);
+      newInputEl.classList.add('edit-input');
+      taskDescEl.replaceWith(newInputEl);
+      // Focus on the input field
+      newInputEl.focus();
+      // Add event listener to the input field to handle the edit action
+      newInputEl.addEventListener('keydown', (event) => {
+        if (event.key === 'Enter') {
+          // Save the edited todo
+          todos[index].description = newInputEl.value;
+          toLocalStorage();
+          renderTodos();
+        }
+      });
+    };
+
+    // Add event listener to each edit icon
+    const editIcon = document.getElementsByClassName('edit');
+    for (let i = 0; i < editIcon.length; i += 1) {
+      editIcon[i].addEventListener('click', () => {
+        handleEdit(i);
+      });
+    }
+    // Function to handle delete action
+    const handleDelete = (index) => {
+      todos.splice(index, 1);
+      toLocalStorage();
+      renderTodos();
+    };
```

Comment:
- [ ] When a todo is deleted, you are to update the indexes of the remaining todos to be in order as they are. 
In the screenshot below, I entered 6 todos, and I deleted them randomly, but the index is not organized. You're required to organize the order. In short, the indexes need to update to be in order 1...2...3.

<img width="415" alt="Screenshot 2023-04-08 at 5 26 27 PM" src="https://user-images.githubusercontent.com/50941074/230734897-25c8c874-2279-473c-83ba-01cd441a5367.png">